browser/components/translation/BergamotTranslator.jsm

```diff
@@ -210,19 +211,26 @@ this.BergamotTranslator.prototype = {
    * @returns string       translated text in the target language
    */
   _getTranslationFromParagraph(paragraph) {
+    // ToDo: Add message-system logging later in this method
     // Each 'Paragraph' contains a list of 'Sentence translation' list.
-    // There is only 1 such list.
+    // There should be only 1 such list.
     let sentenceTranslationList = paragraph[0];
 
-    // 'Sentence translation' list contains only 1 entry which
-    // contains n Best Translations
-    let sentenceTranslation = sentenceTranslationList[0];
-    let nBestTranslations = sentenceTranslation.nBest;
+    // 'Sentence translation' list contains 'Sentence translation' objects
+    // where each object contains all the information related to translation
+    // of each sentence in source language.
+    let translation = "";
+    for (let sentenceTranslation of sentenceTranslationList) {
+      let nBestTranslations = sentenceTranslation.nBest;
 
-    // We request for 1 translation and should get 1 translation back.
-    // Translated text in target language is in 'translation' field.
-    let translation = nBestTranslations[0].translation;
-    return translation;
+      // Depending on the request, there might be multiple 'best translations'.
+      // We are fetching the best one (present in 'translation' field).
+      translation += nBestTranslations[0].translation;
+
+      // Add spaces between individual sentences
+      translation += " ";
```

Comment:
Is using spaces to separate translations language independent?

Reply:
This will always add an extra space at the end of translation. So you should rework this loop so spaces are only between translations and not at the end of a translation.

Also, this final space will disturb the formatting of the translated version of text within a `<pre>` HTML tag.

Reply:
Do we always know that each translation was originally separated by a single space?

For example in a `<pre>` tag two translations might be separated by a new line or many spacaes. So using a single space will disturb the formatting of the translated version of text within a `<pre>` HTML tag.

Reply:
> Do we always know that each translation was originally separated by a single space?
> 
> For example in a `<pre>` tag two translations might be separated by a new line or many spacaes. So using a single space will disturb the formatting of the translated version of text within a `<pre>` HTML tag.

The problem right now is with the rest server. If we send a text with whitespaces with "options" set in the request then it provides translated sentences without retaining these whitespaces. e.g.
**"  Was ist deine Name?    Was brauchst du von mir?\n"**
translates to
**"What is your name? What do you need from me?"**

That is why (for now), I had to append these whitespaces after each sentence.

Reply:
> This will always add an extra space at the end of translation. So you should rework this loop so spaces are only between translations and not at the end of a translation.
> 
> Also, this final space will disturb the formatting of the translated version of text within a `<pre>` HTML tag.

I am handling this case in the line after the loop ends where I trim the leading and trailing whitespaces.

Reply:
> Is using spaces to separate translations language independent?

Actually, I am not sure about this.

Reply:
I think, I can add a comment here mentioning that rest server response doesn't retain whitespace information and we need to change it once this problem is fixed in rest server.
Do you think we can proceed this way?

Reply:
Yeah, I guess that's about as good as it gets for now.